Feature/#24/jckreg/wrap up

charles-moloney opened the pull request:
This PR is a combination of a number of features. In the future, these should be broken up into multiple separate tickets. Right now, we are doing the following:

1. Adding new logic and test functionality to ensure that comments are preserved at all parts of the code when transformed in order to comply with keeping necessary copyright information from 3rd parties
2. Added a new integration test and comparison data to automate the comparison of all transformed files to ensure future changes don't break existing functionality (this is where a lot of the new files are coming from)
3. Modified the README and .project name to better reflect the new name of ARG-V and its structure.
4. Modified gradle build scripts and dataset.csv to be more comprehensive and cover more use cases.
5. Added helpful inline statements during transformation for debugging during execution.
6. Various bugfixes and small modifications

jckreg committed:
basic download main and gradle target

jckreg committed:
all is well?

jckreg committed:
compiling and debugging svcomp benchmarks

jckreg committed:
working on transformationa and compile issues and verbosity

jckreg committed:
removing unneeded imports

jckreg committed:
simplifying transform all and logging logic

jckreg committed:
updating readme

jckreg committed:
updating readme

jckreg committed:
updating readme and build

jckreg committed:
merging in new readme, regression test and configs. Adding verifier

jckreg committed:
Merge branch 'feature/#0/jckreg/download_only' into feature/#24/jckreg/wrapUp

jckreg committed:
merging feature/28/jckreg/failed_to_compile

jckreg committed:
Merge branch 'feature/#24/jckreg/clean_up' into feature/#24/jckreg/wrapUp

jckreg committed:
cleaning build and editing readme

swflint committed:
Make a handful of formatting changes to the readme

swflint committed:
Continue formatting changes

swflint committed:
Make some additional formatting changes & fix some typos

charles-moloney committed:
Modifying Transformer Main to be more inline with SV-COMP format

charles-moloney committed:
Adding additional property to .yml files

charles-moloney committed:
Renaming regression tests to integration

charles-moloney committed:
Implementing new tests

charles-moloney committed:
Additional Unit Tests

charles-moloney committed:
Updating comments test

charles-moloney committed:
Keeping comments preserved and adding QoL test

swflint commented on .project:
```diff
@@ -1,36 +1,17 @@
-			</arguments>
-		</buildCommand>
-		<buildCommand>
-			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
```
Does this get rid of the gradle builder & switch to something not gradle-aware?

charles-moloney replied:
I mistakenly committed this. Changing back in next push

swflint commented on README.md:
```diff
@@ -1,110 +1,199 @@
+Other tasks are available and can be seen by running `./gradlew tasks` from
+the command line. This will display all tasks available and a brief description
+of what they do.
+<!--## Using the Tool-->
```
The comments seem unnecessary.  Could we remove them?

charles-moloney replied:
Going to defer to @jckreg here because he made the changes. I am personally in favor of a longer README as it's the first thing anyone in the repo looks at

jckreg replied:
I am good either way. I feel like the README still needs work done on it at the moment and would take any feedback, comments or complaints that people have on the current for to get it to a better position. We can remove these if they feel redundant. They are currently not showing in the read section of github onnly in the source code. Would you rather remove them entirely or rework them?

swflint commented on integrationExpected/ComplexSymbolicIntTestCase.yml:
Generally, for the expected test results, could we put them somewhere other than top-level?  Maybe a `test/expectedData` or something?

charles-moloney committed:
Moving integration tests resource and refactoring test suite

integration tests now live in the test folder; changed to parameterized tests so they have more accurate error messages

charles-moloney committed:
Correcting build script so pipeline works

charles-moloney committed:
Re-correcting build.gradle.kts

charles-moloney committed:
Updating sourcesets again

charles-moloney committed:
Adding additional exclusion to tests task

charles-moloney committed:
Correcting exclusion language for gradle 9.0

charles-moloney committed:
Correcting for new sourcesets

charles-moloney committed:
correcting source sets

charles-moloney committed:
making path name more ubiquitious

charles-moloney committed:
Update builNTest.yaml

setting test to show debug output

charles-moloney committed:
Update builNTest.yaml

moving debug output for testing

charles-moloney committed:
Update builNTest.yaml

Adding test log

charles-moloney committed:
Update builNTest.yaml

Changing artifact upload to v4

charles-moloney committed:
Update builNTest.yaml

Grabbing all artifacts

charles-moloney committed:
Trimming files and attempting test with debug output

swflint commented on .github/workflows/builNTest.yaml:
```diff
@@ -22,6 +22,7 @@ jobs:
           distribution: zulu
       - name: Set Up App
         run: |
+          ./gradlew test --debug
```
Shouldn't this be *after* cleaning and building?

charles-moloney replied:
This was for my own debugging purposes. Removing now!

charles-moloney committed:
Fixing recursive depth issue with files.walk

charles-moloney committed:
Update builNTest.yaml

Removing unnecessary test workflow

jckreg commented on src/java/transform/visitors/TransformVisitor.java:
```diff
@@ -283,6 +278,58 @@ public void endVisit(CompilationUnit node) {
+	        
+			//String[] importSplit = importName.split("\\.");
+			//String className = importSplit[importSplit.length - 1];
+		//	if (!importName.startsWith("java.") && !importName.startsWith("javax.")) {
```
Are we thinking we may need these commented out lines in the future?

swflint replied:
@charles-moloney?

jckreg commented:
Everything looks good on the changes. I would like to see the README be reworked but think that will need to be a team effort if we wanted to start a discussion and get other feedback I think that would be great but don't think it should hold back this PR. We can fix in the future as needed.

I am however getting failed tests on my local run on the Symbolic Double, Float and Integer tests as well as on the PreserveComments test. I will run run them again and see if there is something I did wrong. If anyone else has had issues running locally please let us know.

jckreg commented:
I resolved my local failed tests issues. It was the java distro and or version that I was using. Matching the one used by the runner I resolved all issues and have more successful benchmark compilations when running the full program. This is something that should probably be noted in the readme for other developers having issues.

jckreg reviewed (approved):
If we do more fixes on the readme lets do it in another PR and get this in in the meantime. As for the commented out code we can leave it for now until we better determine if it will be needed.
